### ISSUE TITLE
Restore navigational `<link>` relations

### ISSUE THREAD

**Issue by bberberov** [glfs-book/glfs#52]
Would you be interested in restoring the [link relations](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel) that would normally be generated by DocBook XSL in the `<head>` element?  The customizations in [lfs-navigational.xsl](https://github.com/glfs-book/glfs/blame/trunk/stylesheets/lfs-xsl/xhtml/lfs-navigational.xsl#L10-L32) disable them.  I've made patches for this for LFS and BLFS that are still out-of-tree.  I can try and integrate them with your changes.  I've also found that [accesskey](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/accessKey) is not usable in modern browsers and have removed it.

**Comment by zeckma-tech** [comment]
Could you link to those patches in this thread? I can bring this up with the LFS editing team so we can discuss it.

**Comment by bberberov** [comment]
I'll fork your repo and prepare a branch so you can take a look.

You can see how it looks on LFS and BLFS here:

- https://gitlab.com/clu-os/docs/lfs
- https://gitlab.com/clu-os/docs/blfs

In your repo it will be a little different, from what I glanced at last week.

**Comment by zeckma-tech** [comment]
A few things: accesskey does work, and you need to do `alt+shift+<key>`; I applied the change to BLFS locally but don't see any noticeable difference, although rendering the book now pops out a warning; what is the use case for restoring navigational links in `<head>`?

I also brought this up with the editor team for {,B}LFS and we are discussing it.

**Comment by bberberov** [comment]
I see. `Alt`+`Shift`+`<key>` does focus on the appropriate link.  I'll revert my changes.  Thanks.

I pushed this branch with the commit, but I did not create a pull request:

https://github.com/bberberov/glfs/tree/link_rels

It preserves the favicons, and logo on `index.html`, while using the original `html.head` template.

I have created extensions that use the `rel`s to add navigation buttons on the browser toolbar.  They work with DocBook, Docutils, GNU Texinfo, Sphinx, etc. documentation pages, but LFSes strip them.  Plus I think it's OK to use the original `html.head` template if it's not breaking anything.

**Comment by zeckma-tech** [comment]
I think the main problem right now with it is it produces a warning about book/bookinfo not having any leaves or something to that effect. Nothing else seems to be out of the ordinary. This change probably won't make it into {,B}LFS. Will wait on this more but may add the `wontfix` label and close this issue, but we'll see.

**Comment by bberberov** [comment]
I think you're talking about this warning.

```
No template for "/book/bookinfo" (or any of its leaves) exists in the context named "title" in the "en" localization.
```

I've seen it before, but I never traced it, until tonight.

It seems to be caused by the `html.head` template from `docbook-xsl` `1.79.2`, since I removed the simpler `html.head` from `lfs-navigation.xsl` in my commit.

More specifically it's this fragment ...

```xml
...
<!-- up link not valid in HTML5 -->
  <xsl:if test="$up and $div.element != 'section'">
...
      <xsl:attribute name="title">
        <xsl:apply-templates select="$up" mode="object.title.markup.textonly"/>
      </xsl:attribute>
...
  </xsl:if>
...
```

... when looking up the `$up` on the `legalnotice` page.  I know that the `legalnotice` templates have been modified by the LFSes, but I haven't looked into it yet.

To test, you can copy the `html.head` from `chunk-common.xsl` to `lfs-navigation.xsl` and modify it like this:

```xml
...
<!-- up link not valid in HTML5 -->
  <xsl:if test="$up and $this != legalnotice and $div.element != 'section'">
...
      <xsl:attribute name="title">
        <xsl:apply-templates select="$up" mode="object.title.markup.textonly"/>
      </xsl:attribute>
...
  </xsl:if>
...
```

The warning goes away on my system.  The `<link rel='up'>` on `legalnotice` still works, but has an extra hash value, and no title.

Even with this issue, there are several ways to partially implement support for `rel`s, if you don't want to ignore, or workaround, this small issue.  At some point I might look at what's causing the problem on `legalnotice`, but it's not a priority for me.

**Comment by zeckma-tech** [comment]
Sorry. I appreciate your work, but this is a fix I'm not wanting to add into the book currently. Thanks regardless!
